libs/partners/mongodb/tests/integration_tests/test_vectorstores.py

```diff
@@ -28,23 +33,65 @@ class PatchedMongoDBAtlasVectorSearch(MongoDBAtlasVectorSearch):
     def _insert_texts(self, texts: List[str], metadatas: List[Dict[str, Any]]) -> List:
         """Patched insert_texts that waits for data to be indexed before returning"""
         ids = super()._insert_texts(texts, metadatas)
-        timeout = TIMEOUT
-        while len(ids) != self.similarity_search("sandwich") and timeout >= 0:
+        start = monotonic()
```

Comment:
Just a note. There's going to be a bunch of merge conflicts with the Hybrid Search PR [#23237](https://github.com/langchain-ai/langchain/pull/23237). Once we get this one merged, I will sync and update.